pypesto/storage/save_to_hdf5.py

```diff
@@ -58,23 +48,18 @@ def write(self, problem, overwrite: bool = False):
                                     "information about optimization result."
                                     "If you wish to overwrite the file set"
                                     "overwrite=True.")
+            attrs_to_save = [a for a in dir(problem) if not a.startswith('__')
+                             and not callable(getattr(problem, a))]
 
             problem_grp = f.create_group("problem")
             # problem_grp.attrs['config'] = objective.get_config()
-            problem_grp.attrs[self.DIM] = problem.dim
-            problem_grp.attrs[self.DIM_FULL] = problem.dim_full
-
-            write_float_array(problem_grp, self.LB, problem.lb)
-            write_float_array(problem_grp, self.UB, problem.ub)
-            write_float_array(problem_grp, self.LB_FULL, problem.lb_full)
-            write_float_array(problem_grp, self.UB_FULL, problem.ub_full)
-            write_float_array(problem_grp, self.X_FIXED_VALS,
-                              problem.x_fixed_vals)
-            write_int_array(problem_grp, self.X_FIXED_INDICES,
-                            problem.x_fixed_indices)
-            write_int_array(problem_grp, self.X_FREE_INDICES,
-                            problem.x_free_indices)
-            write_string_array(problem_grp, self.X_NAMES, problem.x_names)
+
+            for problem_attr in attrs_to_save:
```

Comment:
I think it is a design question: Do we want to automatically save everything (new implementation), or do we want to control explicitly what we save (old one). No preferences, thus new one fine for me.

Reply:
I'd expect less trouble with just saving everything, and potentially having some exclusion list if things get too big. Otherwise there will be regular issues with new fields be introduced by not saved...

Reply:
this should be covered by tests by now, but agreeing that this is easier to maintain.

Reply:
Yes, wasn't sure about that and wanted to see your comments :)